go/signedexchange/cmd/dump-signedexchange/main.go

```diff
@@ -7,34 +7,92 @@ import (
 	"io"
 	"io/ioutil"
 	"log"
+	"net/http"
 	"os"
+	"strings"
 	"time"
 
 	"github.com/WICG/webpackage/go/signedexchange/structuredheader"
 
 	"github.com/WICG/webpackage/go/signedexchange"
+	"github.com/WICG/webpackage/go/signedexchange/version"
 )
 
+type headerArgs []string
+
+func (h *headerArgs) String() string {
+	return fmt.Sprintf("%v", *h)
+}
+
+func (h *headerArgs) Set(value string) error {
+	*h = append(*h, value)
+	return nil
+}
+
+var latestVersion = string(version.AllVersions[len(version.AllVersions)-1])
+
 var (
-	flagInput     = flag.String("i", "", "Signed-exchange input file")
-	flagSignature = flag.Bool("signature", false, "Print only signature value")
-	flagVerify    = flag.Bool("verify", false, "Perform signature verification")
 	flagCert      = flag.String("cert", "", "Certificate CBOR file. If specified, used instead of fetching from signature's cert-url")
+	flagHeaders   = flag.Bool("headers", true, "Print headers")
+	flagFilename  = flag.String("i", "", "Signed-exchange input file")
 	flagJSON      = flag.Bool("json", false, "Print output as JSON")
+	flagPayload   = flag.Bool("payload", true, "Print payload")
+	flagSignature = flag.Bool("signature", false, "Print only signature value")
+	flagURI       = flag.String("uri", "", "Signed-exchange uri")
+	flagVerify    = flag.Bool("verify", false, "Perform signature verification")
+	flagVersion   = flag.String("version", latestVersion, "Signed exchange version")
+
+	flagRequestHeader = headerArgs{}
 )
 
+func init() {
+	flag.Var(&flagRequestHeader, "requestHeader", "Request header arguments")
+}
+
 func run() error {
-	in := os.Stdin
-	if *flagInput != "" {
-		var err error
-		in, err = os.Open(*flagInput)
+	fi, err := os.Stdin.Stat()
+	if err != nil {
+		return err
+	}
+	var e *signedexchange.Exchange
+	var in io.Reader = nil
+	if *flagFilename != "" { // read sxg from filename
+		in, err = os.Open(*flagFilename)
 		if err != nil {
 			return err
 		}
```

Comment:
`defer in.Close()` here?

Reply:
Not sure exactly what needs to happen here as `io.Reader has no field or method Close`

Reply:
Ah, then how about this:
```
f, err := os.Open(*flagFilename)
if err != nil {
  return err
}
defer f.Close()
in = f
```